commerce_coordinator/apps/commercetools/catalog_info/edx_utils.py

```diff
@@ -64,6 +64,16 @@ def get_edx_payment_intent_id(order: CTOrder) -> Union[str, None]:
         return pmt.interface_id
     return None
 
+# TODO update get_edx_successful_stripe_payment to accommodate this util logic
+# and replace it with that. 
+def get_edx_payment_info(order: CTOrder):
+    for pr in order.payment_info.payments:
+        pmt = pr.obj
+        if pmt.payment_status.interface_code == PAYMENT_STATUS_INTERFACE_CODE_SUCCEEDED \
+        and pmt.interface_id:
+            return pmt.interface_id, pmt.payment_method_info.payment_interface
+    return None, None
```

Comment:
A question on this: Do we have multiple payments here on which we are looping? The first to match our condition is the relevant one? Is there a possibility that more than one payment objects meets our condition?

Reply:
We store all the payments (failed ones) on the CT Order object and hence have to loop to get the successful one.